codegenerator/cli/templates/static/codegen/src/Config.res

```diff
@@ -16,15 +16,45 @@ type syncConfig = {
 
 type hyperSyncConfig = {endpointUrl: string}
 
+let hyperSyncConfigSchema: S.t<hyperSyncConfig> = S.object(s => {
+  endpointUrl: s.field("endpointUrl", S.string),
+})
+
 type hyperFuelConfig = {endpointUrl: string}
 
+let hyperFuelConfigSchema: S.t<hyperFuelConfig> = S.object(s => {
+  endpointUrl: s.field("endpointUrl", S.string),
+})
+
 type rpcConfig = {
   provider: Ethers.JsonRpcProvider.t,
   syncConfig: syncConfig,
 }
 
 type syncSource = HyperSync(hyperSyncConfig) | HyperFuel(hyperFuelConfig) | Rpc(rpcConfig)
 
+let syncSourceSchema = S.union([
+  S.object(s => {
+    s.tag("kind", "HyperSync")
+    HyperSync(s.field("payload", hyperSyncConfigSchema))
+  }),
+  S.object(s => {
+    s.tag("kind", "HyperFuel")
+    HyperFuel(s.field("payload", hyperFuelConfigSchema))
+  }),
+  S.object(s => {
+    s.tag("kind", "Rpc")
+    //Do not share users private rpc details
+    Rpc(Js.Nullable.Null->Utils.magic)
+  }),
```

Comment:
@DZakh maybe you have some wisdom for how this should look? I want to be able to serialize this to share that it is using an RPC without any of the other values associated with the provider (in this case just a URL)

It's obviously not deserializeable with the same schema but currently don't need deserialization